python/kvikio/kvikio/benchmarks/utils.py

```diff
@@ -45,15 +45,15 @@ def pprint_sys_info() -> None:
             dev = pynvml.nvmlDeviceGetHandleByIndex(0)
 
         if dev is not None:
+            gpu_name = f"{pynvml.nvmlDeviceGetName(dev)} (dev #0)"
             try:
-                pynvml.nvmlDeviceGetMigMode(dev)
-            except pynvml.NVMLError:
-                with contextlib.suppress(pynvml.NVMLError):
-                    gpu_name = f"{pynvml.nvmlDeviceGetName(dev)} (dev #0)"
-                    mem_total = format_bytes(pynvml.nvmlDeviceGetMemoryInfo(dev).total)
-                    bar1_total = format_bytes(
-                        pynvml.nvmlDeviceGetBAR1MemoryInfo(dev).bar1Total
-                    )
+                mem_total = format_bytes(pynvml.nvmlDeviceGetMemoryInfo(dev).total)
+            except pynvml.NVMLError_NotSupported:
+                mem_total = "Device has no memory resource"
+            try:
+                bar1_total = pynvml.nvmlDeviceGetBAR1MemoryInfo(handle).bar1Total
+            except pynvml.NVMLError_NotSupported:
+                bar1_total = "Device has no BAR1 support"
```

Comment:
Is it possible for the attribute access to fail? IOW should we add `AttributeError` to the `except`s?

Or can we count on the attributes to be defined assuming the functions return without error?

Reply:
No, I don't think so, those are fixed attributes of the [nvmlBAR1Memory_t struct](https://docs.nvidia.com/deploy/nvml-api/structnvmlBAR1Memory__t.html#structnvmlBAR1Memory__t), the same applies to [nvmlMemory_t](https://docs.nvidia.com/deploy/nvml-api/structnvmlMemory__t.html#structnvmlMemory__t) and [nvmlMemory_v2_t struct](https://docs.nvidia.com/deploy/nvml-api/structnvmlMemory__v2__t.html#structnvmlMemory__v2__t).